app/src/main/res/layout/dialog_singleedittext.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```

Comment:
The ".zip" in the dialog EditText dissapeared. And it seems like there's too much space between the EditText and the buttons. See here the space between the inner text and the buttons: 
<img height="300" src="https://storage.googleapis.com/material-design/publish/material_v_12/assets/0Bzhp5Z4wHba3SGptQmpIUGpFcXc/components-dialogs-sidebyside.png"/>

PS: May be `android:layout_height="wrap_content"`?

Reply:
See b70dfd5.

However I'm sure if I should or able to fix the buttons in the dialog... as it's from the material-dialogs library. Going further may led us end up constructing the whole dialog on our own but giving up the convenience of material-dialogs.

Reply:
@EmmanuelMess

> And it seems like there's too much space between the EditText and the buttons.

This is inherent flaw of `TextInputLayout`, — it reserves space for error message underneath it's EditText. If you remove the space, the buttons will jump around each time error message visibility changes.  IMO, if you want the validation feedback to appear quickly, a bit of extra empty space is the lesser of evils.

Reply:
@Alexander-- Hold my beer.

Reply:
With 92dc754 @EmmanuelMess introduced shrinking space for EditText's error/warning messages when it's not in use.

While I personally have reservation on this behaviour, if everyone think it's OK, I'll be OK with it.